visualize-admin/visualization-tool
# feat: Categorical dashboard filters

Closes #1669

This PR introduces categorical dashboard filters.
- Each categorical filter displays all available dimension values; there's no cascading filters behavior, as every chart can have a different combination of other filters, which makes it impossible to have a "global cascading filter logic"
- Global categorical filters take precedence over local, chart-defined interactive filters, whose UI is disabled when global filters are active
- Cascading interactive filters behavior for chart is disabled when global categorical filters are enabled – in this mode we want to see no data, when a given combination actually yields no data, instead of reloading the chart filters and being out-of-sync with global filters.

## How to test
1. Go to [this link](https://visualization-tool-git-feat-categorical-dashboard-filters-ixt1.vercel.app/en/create/new?cube=https://environment.ld.admin.ch/foen/ubd003701/5&dataSource=Prod).
2. Add two more charts, so that there are three in total.
3. Modify the filters of every chart in a way that there's a different set of filters for every chart.
4. Proceed to layout options.
5. Switch to dashboard / free canvas layout.
6. See that there are four shared filters to be selected.
7. Toggle the filters and see that the select elements appeared on top of the canvas.
8. Use the filters to see that every chart adheres to global filters.
9. Toggle the filters off to see that the chart reverts back to its original filters.
10. You can also experiment with adding interactive filters to charts and seeing that they would be overwritten by global filters and that the state is reverted back to individual interactive filter after disabling global filter.

## To do in the future
- Refactor interactive filters so that they are scoped per cube. Currently if there are dimensions with the same iris across different cubes, we'd combine them and filter as one.
